test/rules/github.js

```diff
@@ -138,3 +138,105 @@ test.serial('github - remote origin is an GitLab repo', async t => {
 		}
 	]);
 });
+
+test.serial('github - invalid token', async t => {
+	const execaStub = sandbox.stub(github.execa, 'stdout');
+	const gotStub = sandbox.stub(github.got, 'get');
+
+	execaStub
+		.withArgs('git', ['config', '--get', 'remote.origin.url'])
+		.returns('git@github.com:sindresorhus/awesome-lint-test.git');
+
+	gotStub
+		.withArgs('https://api.github.com/repos/sindresorhus/awesome-lint-test')
+		.rejects({
+			statusCode: 401
+		});
+
+	const messages = await lint({config, filename: 'test/fixtures/github/0.md'});
+	t.deepEqual(messages, [
+		{
+			ruleId: 'awesome/github',
+			message: 'Unauthorized access or token is invalid'
+		}
+	]);
+});
+
+test.serial('github - API rate limit with token', async t => {
+	const execaStub = sandbox.stub(github.execa, 'stdout');
+	const gotStub = sandbox.stub(github.got, 'get');
+	// eslint-disable-next-line camelcase
+	process.env.github_token = 'abcd';
+
+	execaStub
+		.withArgs('git', ['config', '--get', 'remote.origin.url'])
+		.returns('git@github.com:sindresorhus/awesome-lint-test.git');
+
+	gotStub
+		.withArgs('https://api.github.com/repos/sindresorhus/awesome-lint-test')
+		.rejects({
+			statusCode: 403,
+			headers: {
+				'x-ratelimit-limit': 5000
+			}
+		});
+
+	const messages = await lint({config, filename: 'test/fixtures/github/0.md'});
+	t.deepEqual(messages, [
+		{
+			ruleId: 'awesome/github',
+			message: 'API rate limit exceeded max of 5000 requests per hour'
+		}
+	]);
+
+	delete process.env.github_token;
+});
+
+test.serial('github - API rate limit without token', async t => {
+	const execaStub = sandbox.stub(github.execa, 'stdout');
+	const gotStub = sandbox.stub(github.got, 'get');
+
+	execaStub
+		.withArgs('git', ['config', '--get', 'remote.origin.url'])
+		.returns('git@github.com:sindresorhus/awesome-lint-test.git');
+
+	gotStub
+		.withArgs('https://api.github.com/repos/sindresorhus/awesome-lint-test')
+		.rejects({
+			statusCode: 403,
+			headers: {
+				'x-ratelimit-limit': 60
+			}
+		});
+
+	const messages = await lint({config, filename: 'test/fixtures/github/0.md'});
+	t.deepEqual(messages, [
+		{
+			ruleId: 'awesome/github',
+			message: 'API rate limit exceeded max of 60 requests per hour, use a token to increment the number of requests'
```

Comment:
This could be better written.

```suggestion
			message: 'API rate limit of 60 requests per hour exceeded. Use a personal token to increase the number of requests.'
```